### ISSUE TITLE
Extra indentation causes script not to work

### ISSUE THREAD

**Issue by dc42** [dc42/RepRapFirmware#897]
### Duet Forum Discussion Thread

https://forum.duet3d.com/topic/33363/writing-variables-to-a-file-problems-with-nested-echo-commands

### Which Duet products are you using?

- [ ] Duet2-Wifi
- [ ] Duet2-Ethernet
- [ ] Duet Expansion Breakout Board
- [ ] Duex2
- [ ] Duex5
- [ ] Duet2-Maestro
- [ ] Maestro Dual Driver Expansion
- [ ] Duet3-6HC
- [ ] Duet3-3HC
- [ ] Duet3-1XD
- [ ] Duet3-1LC
- [ ] Duet3-Tool Distribution Board
- [X] Duet3-Mini5+
- [ ] Duet3-Mini2+
- [ ] Raspberry Pi or other SBC
- [ ] SmartEffector
- [ ] Magnetic Filament Sensor
- [ ] Laser Filament Sensor
- [ ] PT100 Daughterboard
- [ ] Thermocouple Daughterboard
- [ ] PanelDue
- [ ] Other
- [ ] None

### Firmware Version

3.5.0-beta.4

### Duet Web Control Version

3.5.0-beta.4

### Are you using a Single Board Computer (RaspberryPi) with your Duet?

- [ ] Yes I use a SBC.
- [X] No I do not use a SBC.

### What happened?

Additional indentation between a M291 command in an if-block and a following nested if-block appears to cause the nested if-block not to be executed. Un-indenting the if-block relative to the M291 command solved it. Inserting an un-indented echo command between the M291 command and the indented if-block also solved it.

**Comment by dc42** [comment]
Simple test script to demonstrate the problem:
```
if true
    M291 S2 P"Press OK"
        echo "got ok"
```
Another example:
```
if true
    M120
    M121
        echo "got ok"
```
In both cases the "got ok" message is not echoed, unless either it is not indented wrt the previous M291 or M121 command, or another echo command that is not indented is inserted before it.

**Comment by dc42** [comment]
Fixed in 3.5-dev branch.
